apollo-federation/src/query_graph/graph_path.rs

```diff
@@ -1865,6 +1841,9 @@ where
             excluded_conditions,
             None,
         )?;
+        if matches!(resolution, ConditionResolution::Unsatisfied { .. }) {
```

Comment:
Bugs:
- We should early return here if the edge's conditions are unsatisfiable.